### PR TITLE
libcoro: add version 0.11.1

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **libcoro/0.11**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libcoro/all/conanfile.py
Comment (uilianries) [inline recipes/libcoro/all/conanfile.py:1]: @toge thank you for updating this project. I helped this project, so I can sure it works on Windows, Mac and Linux. Could you please lift the validation that only works with GCC?

**Comment by uilianries** [comment]
@toge That error is really weird, I'll investigate further. I now more options changed in 0.11 and probably some source is missing when building.

**Comment by uilianries** [comment]
Some weird behavior, when using `cmake_dependent_option` CMake feature with Conan, for some reason, it does not work.

https://c3i.jfrog.io/c3i/misc/logs/pr/22764/2-linux-gcc11/libcoro/0.11//1e83dbe0105b2ee2583fa6fe1489fdb74a950fe7-build.txt

Some options should enabled, they are not:

```
libcoro LIBCORO_EXTERNAL_DEPENDENCIES = ON
libcoro LIBCORO_BUILD_TESTS           = OFF
libcoro LIBCORO_CODE_COVERAGE         = OFF
libcoro LIBCORO_BUILD_EXAMPLES        = OFF
libcoro LIBCORO_FEATURE_NETWORKING    = ON
libcoro LIBCORO_FEATURE_TLS           = ON
libcoro LIBCORO_RUN_GITCONFIG         = OFF
libcoro LIBCORO_BUILD_SHARED_LIBS     = ON
```

Locally, running the same command like, but using only cmake in the docker image, with conanfile.txt, it works.

File: recipes/libcoro/all/conanfile.py [hunk 1]
```diff
@@ -40,8 +40,12 @@ def _min_cppstd(self):
     @property
     def _minimum_compilers_version(self):
         return {
-            "gcc": "10.2.1",
-        }
+                "gcc": "10.2.1",
+                "clang": "16.0.0",
+                "apple-clang": "13",
```
Comment (uilianries) [inline recipes/libcoro/all/conanfile.py:45]:
```suggestion
                "apple-clang": "16",
```

Need to re-check it with the upstream, but probably the project needs an update.

**Comment by uilianries** [comment]
I'll check the sys header missing.

**Comment by uilianries** [comment]
After checking the upstream's code, some options are not available on Windows, but the recipe did not have the logic, because until now we have supported only Linux gcc. I removed dynamically the affected options.

**Comment by uilianries** [comment]
The current error is because shared library was not supported before 0.11 and the custom patch available in CCI is bugged (need extra work due static variables). I'll remove those patches and keep it as static library. Since 0.11 anyone can use shared library with safety (tested in the upstream).

**Comment by toge** [comment]
@uilianries 
Sorry conflicted!
I am goting to revert it.

**Comment by uilianries** [comment]
@toge No, I'm sorry of abusing your PR 😅 The fact is `WINDOWS_EXPORT_ALL_SYMBOLS` will not work here. CMake is capable to export all methods, but not static variables. The libcoro has 2 public static variables that need the dlspec directly in the code. You can see the real fix here: https://github.com/jbaldwin/libcoro/pull/236

**Comment by toge** [comment]
@uilianries 
Thank you for your information!
Now I understand.

**Comment by uilianries** [comment]
The CI is not smart enough to understand dynamic shared option. We will need to keep it, but manage as invalid configuration.

**Comment by uilianries** [comment]
@toge It's finally working! 🎉

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by toge** [comment]
@uilianries 
Thanks a lot!

**Review by jwillikers — APPROVED** [review]

**Pull request merged**